### PR TITLE
Move registration from AssetManager to AssetRegistrar

### PR DESCRIPTION
**Pull request by devanych**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #76

### CODE REVIEW

File: src/AssetRegistrar.php [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+
+/**
+ * AssetRegistrar registers asset files from a bundle considering dependencies.
+ *
+ * @psalm-type CssFile = array{0:string,1?:int}&array
+ * @psalm-type CssString = array{0:mixed,1?:int}&array
+ * @psalm-type JsFile = array{0:string,1?:int}&array
+ * @psalm-type JsString = array{0:mixed,1?:int}&array
+ */
+final class AssetRegistrar
```
Comment (vjik) [inline src/AssetRegistrar.php:28]: Mark class as `@internal`. Class should be used only into package.

File: src/AssetRegistrar.php [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+use function get_class;
+use function gettype;
+use function is_array;
+use function is_int;
+use function is_object;
+use function is_string;
+use function sprintf;
+
+/**
+ * AssetRegistrar registers asset files from a bundle considering dependencies.
```
Comment (vjik) [inline src/AssetRegistrar.php:21]:
```suggestion
 * AssetRegistrar registers asset files, code blocks and variables from a bundle considering dependencies.
```

File: src/AssetManager.php [hunk 3]
```diff
@@ -49,30 +42,9 @@ final class AssetManager
...
-     * @psalm-var JsString[]
-     */
-    private array $jsStrings = [];
-
-    private array $jsVars = [];
-    private ?AssetConverterInterface $converter = null;
     private ?AssetPublisherInterface $publisher = null;
     private AssetLoaderInterface $loader;
+    private AssetRegistrar $registrar;
     private Aliases $aliases;
```
Comment (vjik) [inline src/AssetManager.php:48]: Aliases don't uses in this class, can remove this.

**Review by vjik — CHANGES_REQUESTED** [review]
Look fine for me

File: src/AssetRegistrar.php (hunk 1 above)
Comment (devanych) [reply src/AssetRegistrar.php:30]: Done.

**Review by vjik — APPROVED** [review]

**Review by terabytesoftw — APPROVED** [review]

**Pull request merged**